nncf/common/insertion_point_graph.py

```diff
@@ -243,6 +245,95 @@ def _get_default_post_hook_ip_list(nncf_graph: NNCFGraph) -> List[PostHookInsert
             allowed_post_hook_insertion_points.append(PostHookInsertionPoint(nncf_node.node_name))
         return allowed_post_hook_insertion_points
 
+    def get_input_nodes(self) -> List[str]:
+        """
+        Returns all Input Nodes, nodes having any of INPUT_NOOP_METATYPES.
+
+        :return: All Input Nodes.
+        """
+        output = []
+        for node in self.nodes:
+            if InsertionPointGraph.REGULAR_NODE_REF_NODE_ATTR not in self.nodes[node]:
+                continue
+            if self.nodes[node][InsertionPointGraph.IS_MERGED_NODE_ATTR]:
+                for nncf_node in self.nodes[node][InsertionPointGraph.MERGED_NNCF_NODE_LIST_NODE_ATTR]:
+                    node_k = nncf_node.data[NNCFGraph.KEY_NODE_ATTR]
+                    if self._base_nx_graph.nodes[node_k][NNCFGraph.METATYPE_ATTR] in INPUT_NOOP_METATYPES:
+                        output.append(node)
+                        break
+            else:
+                if self.nodes[node][InsertionPointGraph.REGULAR_NODE_REF_NODE_ATTR].metatype in INPUT_NOOP_METATYPES:
+                    output.append(node)
+        return output
+
+    def get_merged_node_from_single_node_key(self, node_key: str) -> str:
+        """
+        Returns the merged node key if the corresponding node with 'node_key' was merged.
+        Otherwise retuns the 'node_key'.
+
+        :param node_key: The key of the node which is checking on the merged.
+        :return: The merged node key. 'node_key' if it was not merged.
+        """
+        for node in self.nodes:
+            if InsertionPointGraph.REGULAR_NODE_REF_NODE_ATTR not in self.nodes[node]:
+                continue
+            if self.nodes[node][InsertionPointGraph.IS_MERGED_NODE_ATTR]:
+                for nncf_node in self.nodes[node][InsertionPointGraph.MERGED_NNCF_NODE_LIST_NODE_ATTR]:
+                    node_k = nncf_node.data[NNCFGraph.KEY_NODE_ATTR]
+                    if node_key == node_k:
+                        return node
+        return node_key
+
+    def get_all_node_keys(self) -> List[str]:
+        """
+        Returns all keys of all nodes.
+
+        :return: All keys of all nodes.
+        """
+        return list(self.nodes)
+
+    def remove_nodes(self, nodes_keys: List[str]) -> None:
+        """
+        Removes nodes from the graph with corresponding 'node_keys'.
+
+        :param nodes_keys: Node keys to remove.
+        :return: None.
+        """
+        self.remove_nodes_from(nodes_keys)
+
+    def get_next_nodes(self, node_key: str) -> List[str]:
+        """
+        Returns consumer nodes of provided node with key 'node_key'.
+
+        :param node_key: Producer node key.
+        :return: List of consumer nodes of provided node.
+        """
+        return self.succ[node_key]
+
+    def traverse_graph(self,
```

Comment:
Why did you define this method and not reuse it from the `NNCFGraph` that does the same?

Reply:
Done